elasticapm/instrumentation/control.py

```diff
@@ -1,6 +1,18 @@
+import threading
+
 from elasticapm.instrumentation import register
 
 
+_lock = threading.Lock()
+
+
 def instrument():
-    for obj in register.get_instrumentation_objects():
-        obj.instrument()
+    with _lock:
```

Comment:
Is it important that it's thread safe? Seems like you're in a bad situation if you have different threads calling instrument/uninstrument simultaneously

Reply:
The main concern is threaded workers that could all start instrumenting in parallel before the `instrumented` flag is set. You'd end up with multiple layers of wrapping. Granted, we've never heard of such a case, but that's possibly because most people don't uses threaded workers.

Running tests in parallel was another concern (although those are usually run in parallel using processes, not threads, so probably not an issue).